### PR TITLE
update neon reference

### PR DESCRIPTION
**Pull request by frogunder**
### What does this PR do?

Update Neon to 3000

### CODE REVIEW

File: doc/topics/releases/2016.11.9.rst [hunk 1]
```diff
@@ -1297,14 +1297,14 @@ Changelog for v2016.11.8..v2016.11.9
 
 * **ISSUE** `#43223`_: (`rallytime`_) Properly deprecate describe_route_table function in boto_vpc module (refs: `#43445`_)
 
-* **PR** `#43456`_: (`rallytime`_) Add Neon to version list
+* **PR** `#43456`_: (`rallytime`_) Add 3000 to version list
```
Comment (Ch3LL) [inline doc/topics/releases/2016.11.9.rst]: we probably shouldn't update commit messages as those should be set to what they were previously

File: doc/topics/releases/2016.11.9.rst [hunk 2]
```diff
@@ -1297,14 +1297,14 @@ Changelog for v2016.11.8..v2016.11.9
...
   @ *2017-09-12 15:00:27 UTC*
 
   * **PR** `#43445`_: (`rallytime`_) Bump deprecation warning for boto_vpc.describe_route_table (refs: `#43456`_)
 
   * 3c429299f9 Merge pull request `#43456`_ from rallytime/43445_follow_up
 
-  * 35c1d8898d Add Neon to version list
+  * 35c1d8898d Add 3000 to version list
```
Comment (Ch3LL) [inline doc/topics/releases/2016.11.9.rst]: here as well, should be changed back to neon

File: doc/topics/releases/2017.7.3.rst [hunk 3]
```diff
@@ -3938,7 +3938,7 @@ Changelog for v2017.7.2..v2017.7.3
 
     * 3c429299f9 Merge pull request `#43456`_ from rallytime/43445_follow_up
 
-      * 35c1d8898d Add Neon to version list
+      * 35c1d8898d Add 3000 to version list
```
Comment (Ch3LL) [inline doc/topics/releases/2017.7.3.rst]: here too

File: doc/topics/releases/2018.3.1.rst [hunk 4]
```diff
@@ -2123,7 +2123,7 @@ Changelog for v2018.3.0..v2018.3.1
 
   * 5fcda3eff8 Merge branch '2018.3' into testing_jinja_filters_avaiable_via_salt_ssh
 
-  * 0adfee9b11 Updating a couple tests.  Fixing check_whitelist_blacklist to work with PY3 when non-iterables are passed.  Adding warning about lst_avg results being wrong and future updates in Neon.
+  * 0adfee9b11 Updating a couple tests.  Fixing check_whitelist_blacklist to work with PY3 when non-iterables are passed.  Adding warning about lst_avg results being wrong and future updates in 3000.
```
Comment (Ch3LL) [inline doc/topics/releases/2018.3.1.rst]: here as well

File: doc/topics/releases/2018.3.1.rst [hunk 5]
```diff
@@ -2897,14 +2897,14 @@ Changelog for v2018.3.0..v2018.3.1
 
 * **ISSUE** `#46334`_: (`sjorge`_) [2018.3.0rc1] Stacktrace on call to nacl.dec (refs: `#46426`_)
 
-* **PR** `#46720`_: (`rallytime`_) Bump deprecation notices in nacl module & runner to Neon
+* **PR** `#46720`_: (`rallytime`_) Bump deprecation notices in nacl module & runner to 3000
```
Comment (Ch3LL) [inline doc/topics/releases/2018.3.1.rst]: ^

File: doc/topics/releases/2018.3.1.rst [hunk 6]
```diff
@@ -2897,14 +2897,14 @@ Changelog for v2018.3.0..v2018.3.1
...
   @ *2018-03-27 21:15:46 UTC*
 
   * **PR** `#46426`_: (`garethgreenaway`_) [2018.3.0rc1] fixes to nacl module & runner (refs: `#46639`_, `#46720`_)
 
   * 65bb37effd Merge pull request `#46720`_ from rallytime/bump-nacl-deprecation
 
-  * 5102c0310c Bump deprecation notices in nacl module & runner to Neon
+  * 5102c0310c Bump deprecation notices in nacl module & runner to 3000
```
Comment (Ch3LL) [inline doc/topics/releases/2018.3.1.rst]: ^

File: doc/topics/releases/2018.3.3.rst [hunk 7]
```diff
@@ -1296,12 +1296,12 @@ Changelog for v2018.3.2..v2018.3.3
 
   * fb7885315c Wrap ElementTree.tostring to make strs, not bytes
 
-* **PR** `#48824`_: (`rallytime`_) Bump deprecation in win_servermanager state to Neon
+* **PR** `#48824`_: (`rallytime`_) Bump deprecation in win_servermanager state to 3000
```
Comment (Ch3LL) [inline doc/topics/releases/2018.3.3.rst]: ^

File: doc/topics/releases/2018.3.3.rst [hunk 8]
```diff
@@ -1296,12 +1296,12 @@ Changelog for v2018.3.2..v2018.3.3
...
   @ *2018-08-02 18:01:34 UTC*
 
   * 1fb7d9431b Merge pull request `#48824`_ from rallytime/win_servermanager_deprecations
 
-  * 6ef5412528 Bump deprecation in win_servermanager state to Neon
+  * 6ef5412528 Bump deprecation in win_servermanager state to 3000
```
Comment (Ch3LL) [inline doc/topics/releases/2018.3.3.rst]: ^

File: salt/cloud/deploy/bootstrap-salt.sh [hunk 9]
```diff
@@ -1295,7 +1295,7 @@ __gather_system_info() {
 #----------------------------------------------------------------------------------------------------------------------
 # shellcheck disable=SC2034
 __ubuntu_derivatives_translation() {
-    UBUNTU_DERIVATIVES="(trisquel|linuxmint|linaro|elementary_os|neon)"
```
Comment (Ch3LL) [inline salt/cloud/deploy/bootstrap-salt.sh]: this should stay neon and all changes in this file. we should not update the bootstrap script so remove all changes here. There is an ubuntu derivative named neon

File: salt/cloud/deploy/bootstrap-salt.sh [hunk 10]
```diff
@@ -1304,7 +1304,7 @@ __ubuntu_derivatives_translation() {
     linuxmint_19_ubuntu_base="18.04"
     linaro_12_ubuntu_base="12.04"
     elementary_os_02_ubuntu_base="12.04"
-    neon_16_ubuntu_base="16.04"
+    3000_16_ubuntu_base="16.04"
```
Comment (Ch3LL) [inline salt/cloud/deploy/bootstrap-salt.sh]: here too ^

File: salt/grains/core.py [hunk 11]
```diff
@@ -1440,7 +1440,7 @@ def id_():
     'void': 'Void',
     'slesexpand': 'RES',
     'linuxmint': 'Mint',
-    'neon': 'KDE neon',
+    '3000': 'KDE 3000',
```
Comment (Ch3LL) [inline salt/grains/core.py]: this should stay neon as its the name of kde release

File: salt/grains/core.py [hunk 12]
```diff
@@ -1503,11 +1503,11 @@ def id_():
     'Devuan': 'Debian',
     'antiX': 'Debian',
     'Kali': 'Debian',
-    'neon': 'Debian',
```
Comment (Ch3LL) [inline salt/grains/core.py]: here too

File: salt/grains/core.py [hunk 13]
```diff
@@ -1503,11 +1503,11 @@ def id_():
...
+    '3000': 'Debian',
     'Cumulus': 'Debian',
     'Deepin': 'Debian',
     'NILinuxRT': 'NILinuxRT',
-    'KDE neon': 'Debian',
+    'KDE 3000': 'Debian',
```
Comment (Ch3LL) [inline salt/grains/core.py]: and here ^

File: salt/modules/aptpkg.py [hunk 14]
```diff
@@ -1666,7 +1666,7 @@ def get_repo(repo, **kwargs):
     ppa_auth = kwargs.get('ppa_auth', None)
     # we have to be clever about this since the repo definition formats
     # are a bit more "loose" than in some other distributions
-    if repo.startswith('ppa:') and __grains__['os'] in ('Ubuntu', 'Mint', 'neon'):
+    if repo.startswith('ppa:') and __grains__['os'] in ('Ubuntu', 'Mint', '3000'):
```
Comment (Ch3LL) [inline salt/modules/aptpkg.py]: this should go back to neon for same reasons as above, there is an actual kde release named neon

File: salt/modules/aptpkg.py [hunk 15]
```diff
@@ -1744,7 +1744,7 @@ def del_repo(repo, **kwargs):
     '''
     _check_apt()
     is_ppa = False
-    if repo.startswith('ppa:') and __grains__['os'] in ('Ubuntu', 'Mint', 'neon'):
+    if repo.startswith('ppa:') and __grains__['os'] in ('Ubuntu', 'Mint', '3000'):
```
Comment (Ch3LL) [inline salt/modules/aptpkg.py]: and here

File: salt/modules/aptpkg.py [hunk 16]
```diff
@@ -2127,7 +2127,7 @@ def mod_repo(repo, saltenv='base', **kwargs):
     # to ensure no one sets some key values that _shouldn't_ be changed on the
     # object itself, this is just a white-list of "ok" to set properties
     if repo.startswith('ppa:'):
-        if __grains__['os'] in ('Ubuntu', 'Mint', 'neon'):
+        if __grains__['os'] in ('Ubuntu', 'Mint', '3000'):
```
Comment (Ch3LL) [inline salt/modules/aptpkg.py]: ^

File: salt/modules/aptpkg.py [hunk 17]
```diff
@@ -2439,7 +2439,7 @@ def expand_repo_def(**kwargs):
 
     sanitized = {}
     repo = salt.utils.pkg.deb.strip_uri(kwargs['repo'])
-    if repo.startswith('ppa:') and __grains__['os'] in ('Ubuntu', 'Mint', 'neon'):
+    if repo.startswith('ppa:') and __grains__['os'] in ('Ubuntu', 'Mint', '3000'):
```
Comment (Ch3LL) [inline salt/modules/aptpkg.py]: ^

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/cloud/deploy/bootstrap-salt.sh [hunk 18]
```diff
@@ -1304,7 +1304,7 @@ __ubuntu_derivatives_translation() {
     linuxmint_19_ubuntu_base="18.04"
     linaro_12_ubuntu_base="12.04"
     elementary_os_02_ubuntu_base="12.04"
-    neon_16_ubuntu_base="16.04"
+    neon="16.04"
```
Comment (Ch3LL) [inline salt/cloud/deploy/bootstrap-salt.sh]: this should still be `neon_16_ubuntu_base`

Comment (Ch3LL) [reply salt/cloud/deploy/bootstrap-salt.sh]: shouldn't see any updates in bootstrap-salt.sh file unless doing a bootstrap release

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — APPROVED** [review]

**Comment by max-arnold** [comment]
@frogunder @Ch3LL I found a few other issues.

Version numbers should be allocated here:

```
doc/topics/releases/version_numbers.rst:- Neon: ``TBD``
salt/version.py:        'Neon'          : (MAX_SIZE - 99, 0),
```

A couple of Neon features were wrongly attributed to Sodium:

```
% grep -r -E '\.\. version' doc | grep -i sodium

doc/topics/mine/index.rst:.. versionchanged:: Sodium
doc/topics/mine/index.rst:.. versionadded:: Sodium
doc/topics/mine/index.rst:.. versionchanged:: Sodium

% grep -r -E '\.\. version' salt | grep -i sodium

salt/states/azurearm_dns.py:.. versionadded:: Sodium
salt/states/azurearm_dns.py:    .. versionadded:: Sodium
salt/states/azurearm_dns.py:    .. versionadded:: Sodium
salt/states/azurearm_dns.py:    .. versionadded:: Sodium
salt/states/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/mine.py:    .. versionchanged:: Sodium
salt/modules/azurearm_dns.py:.. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/azurearm_dns.py:    .. versionadded:: Sodium
salt/modules/cmdmod.py:        .. versionadded:: Sodium
salt/modules/hashutil.py:    .. versionadded:: Sodium
salt/beacons/cert_info.py:.. versionadded:: Sodium
```

Some features are still attributed to Fluorine (I have no idea when they were actually merged, please check the PRs):

```
% grep -r -E '\.\. version' salt | grep -i -E '(flourine|fluorine)'

salt/grains/nvme.py:.. versionadded:: Flourine
salt/modules/gpg.py:        .. versionadded:: fluorine
salt/modules/debuild_pkgbuild.py:        .. versionadded:: fluorine
salt/modules/debuild_pkgbuild.py:        .. versionadded:: fluorine
```

Lowercase `neon` (although the PR addresses that):

```
% grep -r -E '\.\. version' salt | grep  'neon'

salt/pillar/git_pillar.py:.. versionadded:: neon
salt/engines/fluent.py:.. versionadded:: neon
salt/states/virt.py:        .. versionadded:: neon
salt/states/virt.py:        .. versionadded:: neon
salt/states/virt.py:        .. versionadded:: neon
salt/modules/virt.py:        .. versionadded:: neon
salt/modules/virt.py:        .. versionadded:: neon
salt/modules/virt.py:        .. versionadded:: neon
salt/modules/virt.py:        .. versionadded:: neon
salt/modules/virt.py:    .. versionadded:: neon
```

A couple of `.. versionadded` have no space (not sure how that would be rendered by sphinx):

```
% grep -r -E '\.\.version' salt | grep -i -E 'neon'

salt/modules/virt.py:        ..versionadded:: Neon
salt/modules/virt.py:        ..versionadded:: Neon
```

**Pull request merged**